### PR TITLE
EUI-4679 Date error formatting margin adjustment

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
JIRA https://tools.hmcts.net/jira/browse/EUI-4679

Notes: Date input and text input formatting

### CODE REVIEW

File: projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-date/gov-uk-date.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="govuk-form-group" [ngClass]="{'govuk-form-group--error': errorMessage?.isInvalid}" [formGroup]="formGroup">
+<div id="date-input-container" class="govuk-form-group" [ngClass]="{'govuk-form-group--error': errorMessage?.isInvalid}" [formGroup]="formGroup">
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-date/gov-uk-date.component.html:1]: @johnbenjamin-hmcts - what happens if there is more than one `gov-uk-date` component on a page... maybe a problem with duplicate IDs? How about something that's a bit more dynamic, based on the `config` that is passed in? For example, you could do: `<div [id]="config.id+'-date-input-container'">` or similar. I appreciate you'll have to change the CSS selector in `gov-uk-date.component.scss`.

File: projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-date/gov-uk-date.component.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#date-input-container {
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-date/gov-uk-date.component.scss:1]: @johnbenjamin-hmcts - as above, you might need to change the CSS selector depending on what you do with the `date-input-container` ID above.

File: projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-input/gov-uk-input.component.html [hunk 3]
```diff
@@ -4,12 +4,17 @@
 <xuilib-gov-label [config]="config"></xuilib-gov-label>
 
 <span *ngIf="config.hint" [id]="config.id +'-hint'" class="govuk-hint">
- {{config.hint}}
+  {{config.hint}}
 </span>
 
-<xuilib-gov-uk-error-message [config]="config" [errorMessage]="errorMessage"></xuilib-gov-uk-error-message>
+<xuilib-gov-uk-error-message [config]="config" [errorMessage]="errorMessage" *ngIf="errorMessage"></xuilib-gov-uk-error-message>
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-input/gov-uk-input.component.html:10]: @johnbenjamin-hmcts - probably no harm leaving the `*ngIf` here, but is there ever a scenario where `errorMessage` is provided but is falsy? In other words, would there ever be an empty error message?

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - just a few minor observations.

**Review by RiteshHMCTS — APPROVED** [review]

File: projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-date/gov-uk-date.component.scss [hunk 4]
```diff
@@ -1,5 +1,5 @@
-#date-input-container {
-  margin-bottom: 30px;
+.date-input-container {
+  margin-bottom: 30px !important;
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/gov-ui/components/gov-uk-date/gov-uk-date.component.scss:2]: @johnbenjamin-hmcts - hmm... is there another way of applying the style without using `!important`? It's usually frowned upon to use this override.

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - just the one comment on the CSS style; everything else looks OK.

**Review by Daniel-Lam — APPROVED** [review]
@johnbenjamin-hmcts - nice rework, I see you wrapped the whole date component in another `<div>` with a `date-input-container` class.

**Pull request merged**